pkg/minikube/registry/drvs/docker/docker_test.go

```diff
@@ -0,0 +1,94 @@
+// +build darwin
+
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"fmt"
+	"testing"
+)
+
+type testCase struct {
+	version, expect string
+}
+
+func appendVersionVariations(tc []testCase, v []int, reason string) []testCase {
+	appendedTc := append(tc, testCase{
+		version: fmt.Sprintf("linux-%02d.%02d", v[0], v[1]),
+		expect:  reason,
+	})
+
+	// postfix string for unstable channel or patch. https://docs.docker.com/engine/install/
+	patchPostFix := "20180720214833-f61e0f7"
+
+	vs := fmt.Sprintf("%02d.%02d.%d", v[0], v[1], v[2])
+	appendedTc = append(appendedTc, []testCase{
+		{
+			version: fmt.Sprintf("linux-%s", vs),
+			expect:  reason,
+		},
+		{
+			version: fmt.Sprintf("linux-%s-%s", vs, patchPostFix),
```

Comment:
can we add unit test for Mac Os Docker Versions too ? ensure the output of docker verison on mac os and windows are covered in the unit test